### ISSUE TITLE
Missing Bond Check in `CSModule::reportELRewardsStealingPenalty` Function

### ISSUE THREAD

**Issue by 00xWizard** [shieldify-security/issues#34]
## Severity

Critical 

## Description
The `reportELRewardsStealingPenalty` function in `CSModule` contract lacks a check to ensure that the Node Operator has sufficient bond to cover the penalty imposed for EL rewards stealing. The function attempts to lock an amount equal to the stolen funds plus a fixed fine without verifying if the Node Operator's bond is adequate.

```
accounting.lockBondETH(
    nodeOperatorId,
    amount + EL_REWARDS_STEALING_FINE
);
```
the function assumes that the Node Operator always has enough bonds to cover the penalty, which may not always be the case.

## Impact

multiple issues could arise from the missing bond check:

1. Partial Penalty Application: If the Node Operator's bond is insufficient, only a portion of the intended penalty might be applied, leading to inconsistent and unfair penalty enforcement across different Node Operators.

2. the entire transaction might fail if there's insufficient bond, potentially preventing the application of any penalty and wasting gas on transactions.

3. Inconsistent State: If the transaction doesn't fail but only partially locks funds, it could lead to an inconsistent state where the reported penalty doesn't match the actual locked amount.

4. Reduced Deterrent Effect: The uncertainty in penalty application could reduce the deterrent effect of the penalty system, encouraging node operators malicious behavior. 

## Location of Affected Code
https://github.com/lidofinance/community-staking-module/blob/8ce9441dce1001c93d75d065f051013ad5908976/src/CSModule.sol#L1009

## Recommendation

the fix should start by adding a function to get the available bond for a `nodeoperatorId` first in `ICSAccounting` interface which is imported in `CSMoudle` contract
```
    /**
     * @notice Get the available bond amount in ETH for a given node operator
     * @param nodeOperatorId The ID of the node operator
     * @return The amount of ETH available as a bond
     */
    function getAvailableBondETH(uint256 nodeOperatorId) external view returns (uint256);
```
By adding this function to the interface, we ensure that any contract implementing `ICSAccounting` will provide a way to check for available bonds for a node operator. This will allow us to perform the necessary check in the `reportELRewardsStealingPenalty` function.

and then we apply it to the CSModule contract 

```
function reportELRewardsStealingPenalty(
    uint256 nodeOperatorId,
    bytes32 blockHash,
    uint256 amount
) external onlyRole(REPORT_EL_REWARDS_STEALING_PENALTY_ROLE) {
    _onlyExistingNodeOperator(nodeOperatorId);
    
++    uint256 totalPenalty = amount + EL_REWARDS_STEALING_FINE;
++    uint256 availableBond = accounting.getAvailableBondETH(nodeOperatorId);
    
++    require(availableBond >= totalPenalty, "Insufficient bond for penalty");

++   accounting.lockBondETH(nodeOperatorId, totalPenalty);
    
    emit ELRewardsStealingPenaltyReported(nodeOperatorId, blockHash, amount);

   // Nonce should be updated if depositableValidators change
      // No need to normalize queue due to only decrease in depositable possible
        _updateDepositableValidatorsCount({
            nodeOperatorId: nodeOperatorId,
            incrementNonceIfUpdated: true,
            normalizeQueueIfUpdated: false
        });
    }}
```

**Comment by ShieldifyMartin** [comment]
Will the tx revert if the node operator has sufficient bond to cover the penalty? Please provide a PoC as well!

**Comment by 00xWizard** [comment]
no, the tx will not revert if the node operator has sufficient bond to cover the penalty which is the intended functionality. The `require` statement will pass, and the `accounting.lockBondETH` function will be called successfully, locking the required bond amount, but my report highlights the case if otherwise, the amount isn't sufficient then it's likely that only a portion of the intended penalty might be applied, leading to inconsistent and unfair penalty.

**Comment by 00xWizard** [comment]
## Attack Scenario 

1. Deploy the `CSModule` contract, ensuring that the malicious node operator `NodeOperatorId` has a bond of only 5 ETH, which is insufficient to cover the penalty.

2. The contract does not check if the bond is enough before attempting to lock the funds.

3.  Call the `reportELRewardsStealingPenalty` function with 6 ETH stolen funds and a 4 ETH fine, resulting in a total penalty of 10 ETH.

4.  Since there’s no bond check, the contract attempts to lock 10 ETH even though only 5 ETH is available.
Expected Outcomes

5. The transaction succeeds, locking only 5 ETH. This results in an inconsistent state where the malicious node operator is not fully penalized, which would make the penalty useless

**Comment by anontheauditor** [comment]
Bond lock functionality was designed to block more bond tokens than are currently available. This ensures that any rewards acquired between the lock and the actual burn will also be burned.
